src/wallet/walletdb.cpp

```diff
@@ -66,6 +69,17 @@ const std::unordered_set<std::string> LEGACY_TYPES{CRYPTED_KEY, CSCRIPT, DEFAULT
 // WalletBatch
 //
 
+template <typename P>
+static uint256 PrivKeyChecksum(const CPrivKey& privkey, const P& pubkey)
+{
+    // hash pubkey/privkey to accelerate wallet load
+    std::vector<unsigned char> to_hash;
```

Comment:
While you're touching, maybe a good idea to change to `std::vector<unsigned char, secure_allocator<unsigned char>>`?